torchreid/integration/nncf/compression.py

```diff
@@ -74,131 +52,94 @@ def _load_checkpoint_for_nncf(model, filename, map_location=None, strict=False):
+    from nncf.torch.initialization import PTInitializingDataLoader
+
+    nncf_metainfo = NNCFMetaInfo.get_from_checkpoint(checkpoint_path)
+    if nncf_metainfo and nncf_metainfo.compression_enabled:
+        print(f'Read NNCF metainfo with NNCF config from the checkpoint: nncf_metainfo=\n{pformat(nncf_metainfo)}')
     else:
-        # update it just to be on the safe side
-        nncf_metainfo['nncf_config'] = nncf_config_data
-
-    class ReidInitializeDataLoader(InitializingDataLoader):
-        def get_inputs(self, dataloader_output):
-            # define own InitializingDataLoader class using approach like
-            # parse_data_for_train and parse_data_for_eval in the class Engine
-            # dataloader_output[0] should be image here
-            args = (dataloader_output[0], )
-            return args, {}
+        checkpoint_path = None  # it is non-NNCF model
+        nncf_metainfo = NNCFMetaInfo(compression_enabled=True, nncf_config=nncf_config)
 
     cur_device = next(model.parameters()).device
     print(f'NNCF: cur_device = {cur_device}')
 
     if checkpoint_path is None:
         print('No NNCF checkpoint is provided -- register initialize data loader')
+
+        class ReidInitializeDataLoader(PTInitializingDataLoader):
+            def get_inputs(self, dataloader_output):
+                # define own InitializingDataLoader class using approach like
+                # parse_data_for_train and parse_data_for_eval in the class Engine
+                # dataloader_output[0] should be image here
+                args = (dataloader_output[0],)
+                return args, {}
+
+        def model_eval_fn(model):
+            """
+            Runs evaluation of the model on the validation set and
+            returns the target metric value.
+            Used to evaluate the original model before compression
+            if NNCF-based accuracy-aware training is used.
+            """
+            from torchreid.metrics.classification import evaluate_classification
+
+            if test_loader is None:
+                raise RuntimeError('Cannot perform model evaluation on the validation '
+                                   'dataset since the validation data loader was not passed '
+                                   'to wrap_nncf_model')
+
+            target_metric = 'top1'
+            targets = list(test_loader.keys())
+            use_gpu = True if 'cuda' == cur_device.type else False
+            for dataset_name in targets:
+                domain = 'source' if dataset_name in datamanager_for_init.sources else 'target'
+                print('##### Evaluating {} ({}) #####'.format(dataset_name, domain))
+                cmc, m_ap, norm_cm = evaluate_classification(test_loader[dataset_name]['query'], model,
+                                                             use_gpu=use_gpu, topk=(1, 5))
+
+            top1, top5 = cmc[0], cmc[1]
+            if target_metric == 'top1':
```

Comment:
You assigned the value above and check it here. Can the variable change somehow?

Reply:
Probably, yes